src/libcore/num/mod.rs

```diff
@@ -2170,6 +2192,17 @@ macro_rules! uint_impl {
         /// ```
         #[inline]
         #[stable(feature = "wrapping", since = "1.7.0")]
+        #[cfg(not(stage0))]
+        pub fn overflowing_shl(self, rhs: u32) -> (Self, bool) {
+            (unsafe {
+                intrinsics::unchecked_shl(self, (rhs & ($BITS - 1)) as $SelfT)
```

Comment:
Since you’re already working around this code, it would be nice if you used this to implement `wrapping_shl` and used `wrapping_shl` to implement `overflowing_shl` instead of the inverse that we have today.